src/movies.js

```diff
@@ -1,22 +1,99 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
+
+const movies = require("./data")
+
+function getAllDirectors() {
+  const directors = movies.map(function (film) {
+    return film.director
+    })
+  return directors
+  }
+  console.log(getAllDirectors(movies).length) // => 250
+
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+function getAllDirectorsUniq() {
+  const directors = movies.map(function (film) {
+    return film.director
+    }).reduce(function(a,b){
+      if (a.indexOf(b) < 0 ) a.push(b)
+      return a
+    },[])
+  return directors
+  }
+console.log(getAllDirectorsUniq(movies).length) // => 149
+
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+function howManyMovies() {
+  const dramas = movies.filter(function (film) {
+    return film.genre.indexOf('Drama') !== -1
+      })
+  const dramasOfSpielberg = dramas.filter(function (film) {
+    return film.director === 'Steven Spielberg'
+  })
+  return dramasOfSpielberg.length
+}
+console.log(howManyMovies(movies)) // => 4
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+function scoresAverage() {
+  const sumOfScores = movies.reduce(function (acc, film) { 
+    return acc + film.score
+  }, 0)
+  return (sumOfScores / movies.length).toFixed(2)
+}
+console.log(scoresAverage(movies)) // => 8.31
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore() {}
+function dramaMoviesScore() {
+  const dramas = movies.filter(function (film) {
+    return film.genre.indexOf('Drama') !== -1
+  })
+    const scoreDramas = dramas.reduce(function (acc, film) { 
+      return acc + film.score
+    }, 0)
+  return (scoreDramas / movies.length).toFixed(2)
```

Comment:
You could instead run the `scoreAverage` function on the `dramas` array. It will do the work for you:
`return scoresAverage(dramas)`